dlog/plonk/src/prover.rs

```diff
@@ -97,6 +97,20 @@ unsafe impl<'a, G: AffineCurve + ocaml::FromValue<'a>> ocaml::FromValue<'a> for
     }
 }
 
+fn trim_commitment<C: AffineCurve>(c : PolyComm<C>) -> PolyComm<C> {
+    let mut c = c;
+    let mut i = c.unshifted.len() - 1;
+    let zero = C::zero();
+    loop {
+        if c.unshifted[i] != zero {
+            break;
+        }
+        i -= 1;
+    };
+    c.unshifted.truncate(i + 1);
+    c
```

Comment:
I think it'd be cleaner to write it as:

```rust
for (i, u) in c.unshifted.iter().rev().enumerate() {
  if u != C::zero() {
      return PolyComm {
          unshifted: c.unshifted.truncate(i);
          ..c
      }
  }
}
c
```

Reply:
(where unshifted.truncate is implemented to truncate `i` entries)

Reply:
also, isn't this something we should implement automatically whenever we commit to something? (instead of here as a separate function)

Reply:
Hmm.. you know now that I think about it, I think this is incorrect as it breaks zero-knowledge (it leaks that the top coefficients are zero). Will fix

Reply:
I don't think it leaks anything as these zeros we're trimming are public

Reply:
I guess you could further improve the code as:

```rust
let unshifted = c.unshifted.clone();
let mut truncated: Vec<_> = unshifted.into_iter().rev().skip_while(|&num| num == 0u8).collect();
truncated.reverse();
PolyComm {
  unshifted: truncated;
  ..c
}
```